core/src/test/java/org/elasticsearch/index/mapper/TokenCountFieldMapperTests.java

```diff
@@ -102,6 +102,28 @@ public TokenStreamComponents createComponents(String fieldName) {
         assertThat(TokenCountFieldMapper.countPositions(analyzer, "", ""), equalTo(7));
     }
 
+    public void testCountTokens() throws IOException {
+        // We're looking to make sure that we:
+        Token t1 = new Token();      // Don't count tokens without an increment
+        t1.setPositionIncrement(0);
+        Token t2 = new Token();
+        t2.setPositionIncrement(1);  // Count normal tokens with one increment
+        Token t3 = new Token();
+        t2.setPositionIncrement(2);  // Count funny tokens with more than one increment
+        int finalTokenIncrement = 4; // Count the final token increment on the rare token streams that have them
+        Token[] tokens = new Token[] {t1, t2, t3};
+        Collections.shuffle(Arrays.asList(tokens), random());
+        final TokenStream tokenStream = new CannedTokenStream(finalTokenIncrement, 0, tokens);
+        // TODO: we have no CannedAnalyzer?
+        Analyzer analyzer = new Analyzer() {
+                @Override
+                public TokenStreamComponents createComponents(String fieldName) {
+                    return new TokenStreamComponents(new MockTokenizer(), tokenStream);
+                }
+            };
+        assertThat(TokenCountFieldMapper.countTokens(analyzer, "", ""), equalTo(3));
```

Comment:
Hi @jimczi 
First, thanks for this first review.

Then, concerning your comment : the feature global goal is to have an option to count the number of tokens after analysis, whatever their positions. In that unit test, there are 3 tokens in the stream, so that's why I expect 3 as a result.

Given your comment, maybe the option "enable_position_increments" is misnamed for this feature goal ?